src/protocols/secure_channel/tests/TestCASESession.cpp

```diff
@@ -824,6 +843,67 @@ static void CASE_SessionResumptionStorage(nlTestSuite * inSuite, void * inContex
     }
 }
 
+#if CONFIG_IM_BUILD_FOR_UNIT_TEST
```

Comment:
If this approach is not desired, alternatively I could make `TestCASESessionExtra.cpp` where these test are added and have it such that the file is not compiled on known device targets.